src/robotFactory.js

```diff
@@ -1,10 +1,76 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = {};
+    this.coords.x = coords.x || 0;
+    this.coords.y = coords.y || 0;
```

Comment:
you can unite it
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0
   };
```

OR 

```suggestion
   this.coords = {
      x: 0,
      y: 0,
      ...coords,
    }
```